tools/trap/dumpParser.py

```diff
@@ -0,0 +1,622 @@
+
+		utils.setup_symbol_table(bin_path + "System.map", 0)
+
+		# Kernel text start & end addresses in FLASH & RAM regions
+		self.g_stext_flash = utils.get_address_of_symbol("_stext_flash")
+		self.g_etext_flash = utils.get_address_of_symbol("_etext_flash")
+		self.g_stext_ram = 0
+		self.g_etext_ram = 0
+		if (have_ram_kernel_text):
+			self.g_stext_ram = utils.get_address_of_symbol("_stext_ram")
+			self.g_etext_ram = utils.get_address_of_symbol("_etext_ram")
+
+
+		# Check for ELF file existence, if exists open the ELF (tinyara) file in binary format
+		# As well as open using NM utility so that, we can read the symbols easily
+		if self.elf is not None:
+			self.elf_file_fd = open(elf, 'rb')
+			if not self.elf_file_fd:
+				print(('Failed to open {0}'.format(elf)))
+				return None
+
+		# Check for ramdump file existence, if exists open the dump file in binary format
+		if self.dump_file is not None:
+			self.dump_file_fd = open(dump_file, 'rb')
+			if not self.dump_file_fd:
+				print(('Failed to open {0}'.format(dump_file)))
+				return None
+
+			#Split the dump_file with delimiters _ and . to read the start and end addr
+			# This is required to support any RAM dump size of any board
+			temp = re.split(r'[_.]',dump_file)
+			# Get the Start address and end address from dump file name
+			self.ram_base_addr = int(temp[-3],16)
+			self.ram_end_addr = int(temp[-2],16)
+			print(('self.ram_base_addr {0:x}'.format(self.ram_base_addr)))
+			print(('self.ram_end_addr {0:x}'.format(self.ram_end_addr)))
+
+		# Read the elf header to get the offset of text section and ARM exidx section
+		# These offsets will be used while creating ARM exidx table as well as while reading
+		# the address from ELF file at a particular text address
+		with os.popen('arm-none-eabi-readelf -S ' + elf) as readelf_fd:
+			elfdata = readelf_fd.readlines()
+			for line in elfdata:
+				if '.text' in line:
+					word = line.split()
+					#word 5 contains the offset value
+					self.text_offset = word[5]
+					if debug:
+						print(line)
+						print(self.text_offset)
+
+	def __del__(self):
+		self.elf_file_fd.close() # Close the elf file instance
+
+	# Function to search the instructions present in stack address.
+	def stack_lookup(self, addr):
+		if(addr is None):
+			return ('(Invalid address)', 0x0)
+		low = 0
+		high = len(self.stack_table)
+		mid = (low + high) >> 1
+		premid = 0
+
+		while(low <= high):
+
+			if(addr == self.stack_table[mid][0]):
+				return self.stack_table[mid][1]
+
+			if(addr < self.stack_table[mid][0]):
+				high = mid - 1
+
+			else:
+				low = mid + 1
+
+
+			mid = (high + low) >> 1
+
+		print("Sorry. Given address does not present in stack lookup")
+		return None
+
+	# Function to Parse the i/p log file (which contains stackdump during assert) passed as part of -t argument ( No dump is supplied )
+	def parse_log_file(self,log_file,debug=False):
```

Comment:
There is something wrong here. Parsing log file is only supposed to be done in log parser. Not in dump parser. Dump parser is supposed to be used only when log file is not given. Can you pls recheck this?

Reply:
So, Earlier dump parser was also responsible for parsing the log file. But now, for call stack details, both dump parser and log parser are parsing the log file. Is it feasible to remove this ```parse_log_file``` API as we are already parsing in log parser? But I don't know about earlier arch (armv7-r), whether they support log Parsing from ```logParser``` class.